vault/core.go

```diff
@@ -2093,6 +2093,49 @@ func (c *Core) emitMetrics(stopCh chan struct{}) {
 	writeTimer := time.Tick(c.counters.syncInterval)
 	identityCountTimer := time.Tick(time.Minute * 10)
 
+	// The gauge collection processes are started and stopped here
+	// because there's more than one TokenManager created during startup,
+	// but we only want one set of gauges.
+	if c.MetricSink().GaugeInterval != time.Duration(0) {
+		tokenCount1, err := c.MetricSink().NewGaugeCollectionProcess(
```

Comment:
This is the pattern that needs to be copied for each of the newly introduced gauges.

I tried refactoring but it was something of a mess because of the need to use a different function, on whatever object (tokenStore) is hooked up to the core, at the time we collect the metrics.